packages/babel-parser/src/parser/location.js

```diff
@@ -165,18 +163,16 @@ export const Errors = Object.freeze({
   UnexpectedTokenUnaryExponentiation:
     "Illegal expression. Wrap left hand side or entire exponentiation in parentheses.",
   UnsupportedBind: "Binding should be performed on object property.",
-  //todo: rephrase this error message as it is too subjective
   UnsupportedDecoratorExport:
-    "You can only use decorators on an export when exporting a class",
+    "A decorated export must export a class declaration",
   UnsupportedDefaultExport:
     "Only expressions, functions or classes are allowed as the `default` export.",
   UnsupportedImport: "import can only be used in import() or import.meta",
   UnsupportedMetaProperty: "The only valid meta property for %0 is %0.%1",
-  //todo: remove Stage 2 as we are likely to forget updating when it progressed
   UnsupportedParameterDecorator:
-    "Stage 2 decorators cannot be used to decorate parameters",
+    "Decorators cannot be used to decorate parameters",
```

Comment:
Actually, `legacy` decorators can be used to decorate parameters.

Reply:
I think it is okay because this error is not visible to users if they are using `legacy` decorators. Like we did in "Legacy octal literals are not allowed in strict mode", if an error is thrown on a legacy elements, we should prefix its name. Otherwise we are referring to the latest proposal.